src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs

```diff
@@ -393,15 +507,14 @@ private async void ToggleSheetVisibility()
                 return;
             }
 
-            if (m_sheetView == null)
-            {
-                m_sheetView = new SheetView(this);
-            }
-
             if (IsOpen)
             {
+                OnBeforeOpenCommand?.Execute(OnBeforeOpenCommandParameter);
+                OnBeforeOpen?.Invoke(this, new EventArgs());
+
+                m_sheetView = new SheetView(this);

```

Comment:
Keep local variable inside 1 scope.

Reply:
What variablel? `m_sheet` is global and should be used for callback methods like `BeforeRemoval` etc.